ZMQ - issue building on windows

Since possibly an update of ZMQ - on windows, building libxayagame fails with:

```
error -pedantic -Wno-deprecated -DGLOG_NO_ABBREVIATED_SEVERITIES -MT libxayagame_la-transactionmanager.lo -MD -MP -MF .deps/libxayagame_la-transactionmanager.Tpo -c transactionmanager.cpp -o libxayagame_la-transactionmanager.o >/dev/null 2>&1
mv -f .deps/libxayagame_la-transactionmanager.Tpo .deps/libxayagame_la-transactionmanager.Plo
In file included from zmqsubscriber.cpp:7:
zmqsubscriber.cpp: In member function 'bool xaya::internal::ZmqSubscriber::ReceiveMultiparts(std::string&, std::string&, uint32_t&)':
zmqsubscriber.cpp:91:32: error: 'bool zmq::detail::socket_base::recv(zmq::message_t*, int)' is deprecated: from 4.3.1, use recv taking a reference to message_t and recv_flags [-Werror=deprecated-declarations]
  91 |       CHECK (socket->recv (&msg));
     |                                ^
In file included from zmqsubscriber.hpp:11,
                from zmqsubscriber.cpp:5:
C:/msys64/mingw64/include/zmq.hpp:1267:10: note: declared here
1267 |     bool recv(message_t *msg_, int flags_ = 0)
     |          ^~~~
cc1plus.exe: all warnings being treated as errors
make[3]: *** [Makefile:1012: libxayagame_la-zmqsubscriber.lo] Error 1
make[3]: Leaving directory '/home/building/libxayagame/xayagame'
make[2]: *** [Makefile:739: all] Error 2
make[2]: Leaving directory '/home/building/libxayagame/xayagame'
make[1]: *** [Makefile:439: all-recursive] Error 1
make[1]: Leaving directory '/home/building/libxayagame'
make: *** [Makefile:371: all] Error 2
```

--- a/README.md
+++ b/README.md
@@ -37,8 +37,11 @@ for the configuration and/or build to be successful:
   are not fresh enough.  They need to be built from source;
   in particular, it must be a version that includes the commit
   [`4f24acaf4c6737cd07d40a02edad0a56147e0713`](https://github.com/cinemast/libjson-rpc-cpp/commit/4f24acaf4c6737cd07d40a02edad0a56147e0713).
-- [`ZeroMQ C++ bindings`](http://zeromq.org/bindings:cpp):
-  Available in the Debian package `libzmq3-dev`.
+- [`ZeroMQ`](https://zeromq.org/) with
+  [C++ bindings](https://github.com/zeromq/cppzmq):
+  Core ZeroMQ is available in the Debian package `libzmq3-dev`.  The C++
+  bindings need to be installed from the source repository, as the version
+  in the Debian package (at least for Debain 10 "Buster") is too old.
 - [`zlib`](https://zlib.net):
   Available in Debian as `zlib1g-dev`.
 - [SQLite3](https://www.sqlite.org/) with the

--- a/configure.ac
+++ b/configure.ac
@@ -41,7 +41,9 @@ AX_PKG_CHECK_MODULES([GLOG], [], [libglog])
 # make sure we have at least this version.
 AX_PKG_CHECK_MODULES([ZLIB], [], [zlib >= 1.2.11])
 AX_PKG_CHECK_MODULES([OPENSSL], [], [openssl])
-AX_PKG_CHECK_MODULES([ZMQ], [], [libzmq])
+# We use the recv variant taking message_t& over deprecated older functions,
+# which requires at least version 4.3.1.
+AX_PKG_CHECK_MODULES([ZMQ], [], [libzmq >= 4.3.1])
 
 # Private dependencies that are not needed for libxayagame, but only for
 # the unit tests and the example binaries.

--- a/xayagame/zmqsubscriber.cpp
+++ b/xayagame/zmqsubscriber.cpp
@@ -88,7 +88,7 @@ ZmqSubscriber::ReceiveMultiparts (std::string& topic, std::string& payload,
   for (unsigned parts = 1; ; ++parts)
     {
       zmq::message_t msg;
-      CHECK (socket->recv (&msg));
+      CHECK (socket->recv (msg));
 
       switch (parts)
         {

--- a/xayagame/zmqsubscriber_tests.cpp
+++ b/xayagame/zmqsubscriber_tests.cpp
@@ -99,8 +99,12 @@ protected:
     for (size_t i = 0; i < parts.size (); ++i)
       {
         zmq::message_t msg(parts[i].begin (), parts[i].end ());
-        const bool hasMore = (i + 1 < parts.size ());
-        ASSERT_TRUE (sock.send (msg, hasMore ? ZMQ_SNDMORE : 0));
+        zmq::send_flags flags;
+        if (i + 1 < parts.size ())
+          flags = zmq::send_flags::sndmore;
+        else
+          flags = zmq::send_flags::none;
+        ASSERT_TRUE (sock.send (std::move (msg), flags));
       }
   }
 

